include/boost/math/quadrature/naive_monte_carlo.hpp

```diff
@@ -164,9 +164,9 @@ class naive_monte_carlo
         Real inv_denom = 1/static_cast<Real>(((gen.max)()-(gen.min)()));
 
         m_num_threads = (std::max)(m_num_threads, static_cast<uint64_t>(1));
-        m_thread_calls.reset(new std::atomic<uint64_t>[threads]);
-        m_thread_Ss.reset(new std::atomic<Real>[threads]);
-        m_thread_averages.reset(new std::atomic<Real>[threads]);
+        m_thread_calls = std::make_unique<std::atomic<uint64_t>[]>(threads);
+        m_thread_Ss = std::make_unique<std::atomic<Real>[]>(threads);
+        m_thread_averages = std::make_unique<std::atomic<Real>[]>(threads);
```

Comment:
This change will break compatibility with C++11. We will not be moving to C++14 until Boost 1.82